onnxruntime/core/framework/provider_bridge_ort.cc

```diff
@@ -187,12 +198,23 @@ struct Provider_IExecutionProvider_Router_Impl : Provider_IExecutionProvider_Rou
     return std::make_shared<Provider_IAllocator_Impl>(IExecutionProvider::GetAllocator(id, mem_type));
   }
 
+  AllocatorPtr GetAllocator(int id, OrtMemType mem_type) const override {
+    auto allocator = outer_->Provider_GetAllocator(id, mem_type);
+    if (!allocator)
+      return nullptr;
+    return static_cast<Provider_IAllocator_Impl*>(allocator.get())->p_;
```

Comment:
>static_cast<Provider_IAllocator_Impl*> [](start = 11, length = 38)

Are you casting to base or to an unrelated class?

Reply:
It's a base class, as the pointer is a Provider_IAllocator, but this one is always implemented by a Provider_IAllocator_Impl.